eigenda/certificate.go

```diff
@@ -0,0 +1,175 @@
+package eigenda
+
+import (
+	"fmt"
+	"math/big"
+
+	eigenda_common "github.com/Layr-Labs/eigenda/api/grpc/common"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/Layr-Labs/eigenda/api/grpc/disperser"
+
+	cv_binding "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDACertVerifier"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// EigenDAV1Cert is an internal representation of the encoded cert commitment (i.e, disperser.BlobInfo)
+// read from EigenDA proxy. It is used for type compatibility with the Solidity V1 certificate.
+// This object is encoded into to txs submitted to the SequencerInbox.
+type EigenDAV1Cert struct {
+	BlobVerificationProof cv_binding.BlobVerificationProof `json:"blobVerificationProof"`
+	BlobHeader            cv_binding.BlobHeader            `json:"blobHeader"`
+}
+
+/*
+Unlike 4844 there's no need to inject a version byte into the 0th offset of the hash.
+Taking the hash of commitment + length is key to ensure no trust assumption on the data length
+for one-step proving.
+*/
+func (e *EigenDAV1Cert) PreimageHash() (*common.Hash, error) {
```

Comment:
receiver names aren't consistent for this struct's methods

Reply:
Awk - will address this in a follow up PR